### PR TITLE
Flatten Namespaces for Evaluation SDK

### PR DESCRIPTION
**Pull request by needuv**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by nagkumar91 — APPROVED** [review]

File: sdk/evaluation/azure-ai-evaluation/tests/e2etests/test_builtin_evaluators.py
Comment (ninghu) [inline sdk/evaluation/azure-ai-evaluation/tests/e2etests/test_builtin_evaluators.py:30]: I understand this is for internal use, but why the namespace contains double "_eci"?

**Review by ninghu — APPROVED** [review]

File: sdk/evaluation/azure-ai-evaluation/CHANGELOG.md
Comment (ninghu) [inline sdk/evaluation/azure-ai-evaluation/CHANGELOG.md:10]: Not related to this PR, but wonder if we should port the latest change log entry from pf-evals to here? I am referring to v0.3.3 (Upcoming)

https://github.com/microsoft/promptflow/blob/main/src/promptflow-evals/CHANGELOG.md

File: sdk/evaluation/azure-ai-evaluation/tests/e2etests/test_builtin_evaluators.py
Comment (needuv) [reply sdk/evaluation/azure-ai-evaluation/tests/e2etests/test_builtin_evaluators.py:30]: I think it's because the folder and filename where the evaluator is available are both `_eci`

Comment (ninghu) [reply sdk/evaluation/azure-ai-evaluation/tests/e2etests/test_builtin_evaluators.py:30]: @MilesHolland: is this evaluator for 1P customer? should we use this chance to fix this?

Comment (ninghu) [reply sdk/evaluation/azure-ai-evaluation/tests/e2etests/test_builtin_evaluators.py:30]: Would be better change it to from azure.ai.evaluation._eci. import ECIEvaluator?

Comment (nagkumar91) [reply sdk/evaluation/azure-ai-evaluation/tests/e2etests/test_builtin_evaluators.py:30]: We can probably do that in a separate PR. I'm waiting for this to go in, so I can fix the merge conflicts for the simulator.

**Review by diondrapeck — APPROVED** [review]

**Comment by needuv** [comment]
Tests were previously passing, looks like the issue on macos is just a flaky test thing. Will override and merge to unblock other folks

**Pull request merged**
